interp/notation_ops.ml

```diff
@@ -187,7 +187,7 @@ let apply_cases_pattern ?loc (ids_disjpat,id) c =
 let glob_constr_of_notation_constr_with_binders ?loc g f ?(h=default_binder_status_fun) e nc =
   let lt x = DAst.make ?loc x in lt @@ match nc with
   | NVar id -> GVar id
-  | NApp (a,args) -> let e = h.no e in GApp (f e a, List.map (f e) args)
+  | NApp (a,args) -> let e = h.no e in DAst.get (mkGApp (f e a) (List.map (f e) args))
```

Comment:
Couldn't this class of bugs be prevented by using an abstract type (with a view), and smart constructors, like we do for `constr`?

Reply:
Why not, but that would lead the PR much further that its intent, so not in this PR.